source/exe/hot_restart.h

```diff
@@ -138,8 +138,12 @@ class HotRestartImpl : public HotRestart,
   };
 
   struct RpcBase {
-    RpcBase(RpcMessageType type, uint64_t length = sizeof(RpcBase))
-        : type_(type), length_(length) {}
+    RpcBase(RpcMessageType type, uint64_t length = sizeof(RpcBase)) {
+      static_assert(std::is_standard_layout<RpcBase>::value, "is_standard_layout<RpcBase>");
```

Comment:
Yeah, this code basically never changes, so boilerplate is fine (I assume you mean just include an RpcBase struct at the beginning of each "derived" class). The other option is to have a template allocator that allocates a derived RPC, memset it to 0, and then does placement new. I'm fine with whatever solution, I would just really prefer to not have to rev the RPC protocol version if possible.

If you do change the layout to get rid of the derivation, can you just spot check that sizeof() does not change? Otherwise we still have to rev the version.